### PR TITLE
feat: add forgot password email

### PR DESCRIPTION
**Pull request by dacgray**
We want to switch the forgot password email in shop to mandrill..

### HTML Test

image

## Text Test

image

### Legacy

image

### CODE REVIEW

File: src/Mandrill/Emails/ForgotPasswordVerificationEmail.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        $this->translator = $lang;
+
+        $this->setSubject($this->translator->translate(self::SUBJECT))
+            ->addRecipient($recipientEmail, $recipientName)
+            ->addGlobalMergeVar(self::GLOBAL_MERGE_VAR_LINK, $passwordResetLink)
+            ->addGlobalMergeVar(self::GLOBAL_MERGE_VAR_TEXT_FOOTER, $this->getTextFooter())
+            ->addTemplateContent('footer', $this->getHtmlFooter());
+
+        foreach (self::GLOBAL_MERGE_VARS as $var) {
+            $this->addGlobalMergeVar($var['name'], $this->translator->translate($var['content']));
```
Comment (turboatcrazyfactory) [inline src/Mandrill/Emails/ForgotPasswordVerificationEmail.php]: Not sure if we need to pass `$var['content']` or `$var['name']` to `->translate()`.

File: tests/Unit/Mandrill/Emails/ForgotPasswordVerificationEmailTest.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * @throws \Throwable
+     */
+    public function testConstructor()
+    {
+        $messagesMock = $this->createMock(\Mandrill_Messages::class);
+        $messagesMock->expects($this->once())
+            ->method('sendTemplate')
+            ->willReturnCallback(function (
+                $template_name,
```
Comment (turboatcrazyfactory) [inline tests/Unit/Mandrill/Emails/ForgotPasswordVerificationEmailTest.php]: mixed use of camelCase and snake_case ?

File: src/Mandrill/Emails/LegacyFooters.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ */
+trait LegacyFooters
+{
+    /**
+     * @return string
+     */
+    protected function getHtmlFooter(): string
+    {
+        return <<<html
+            <table
```
[the remark this answers is not in the document]
Comment (dacgray) [reply src/Mandrill/Emails/LegacyFooters.php]: hmm, yeah, i was thinking about that.  

I settled on this dirty solution for a few reasons:
- I need to use `{$this->translate('bla')}` in the html string
- This is legacy crap that we'll throw out once we re-do the email templates
- It's nice to have all in one place

Moving into a separate file would mean using a template engine which becomes overkill.

File: src/Mandrill/Emails/ForgotPasswordVerificationEmail.php (hunk 1 above)
Comment (dacgray) [reply src/Mandrill/Emails/ForgotPasswordVerificationEmail.php]: Just the content.

I'll add some comments to the constant.

`name` is the template key in the template residing on Mandrill.  `content` is the translation key we have in our *.json translation files.

**Pull request merged**